### PR TITLE
Add include directories for OpenTimeConfig.cmake/OpenTimelineIOConfig…

### LINKED ISSUE #915: Add include directories to OpenTimeConfig.cmake/OpenTimelineIOConfig.cmake
The auto-generated OpenTimeConfig.cmake/OpenTimelineIOConfig.cmake files are missing the include directories for the OTIO headers which makes compiling C++ programs difficult. This can be fixed by adding the "INCLUDES" argument to the install command for OpenTime/OpenTimelineIO. I'll submit a PR with the proposed change...

### PR DESCRIPTION
**Pull request by darbyjohnston**
```Fixes #915```

This change adds the include directories to the auto-generated OpenTimeConfig.cmake/OpenTimelineIOConfig.cmake files so that C++ projects using OTIO can find the header files.

### CODE REVIEW

**Review by meshula — CHANGES_REQUESTED** [review]
Thanks for catching this! One little twist to address...

File: src/opentimelineio/CMakeLists.txt [hunk 1]
```diff
@@ -109,6 +109,7 @@ if(OTIO_CXX_INSTALL)
 
     install(TARGETS opentimelineio
            EXPORT OpenTimelineIOConfig
+           INCLUDES DESTINATION "${OTIO_RESOLVED_CXX_INSTALL_DIR}/include;${OTIO_RESOLVED_CXX_INSTALL_DIR}/include/opentimelineio/deps"
```
[the remark this answers is not in the document]
Comment (meshula) [reply src/opentimelineio/CMakeLists.txt:112]: Also, not related to your change, just thinking out loud, we haven't considered "what if the user wants our any, but not our optional, or vice versa?" I hope we can remove these dependencies before it actually becomes an issue!

[the remark this answers is not in the document]
Comment (darbyjohnston) [reply src/opentimelineio/CMakeLists.txt]: Good point, I added a conditional for OTIO_DEPENDENCIES_INSTALL.

[the remark this answers is not in the document]
Comment (darbyjohnston) [reply src/opentimelineio/CMakeLists.txt]: The documentation mentions that any/optional are provided in C++17? If CMAKE_CXX_STANDARD was bumped up to 17 they could be removed...

**Review by meshula — APPROVED** [review]

**Comment by meshula** [comment]
Thanks!

**Pull request merged**